File: Python/O(N)/on_36.py
# Python3 program for the above approach 
  
# Function to left rotate 
def left_rotate(arr):
      
    last = arr[1]; 
    for i in range(3, len(arr), 2):
        arr[i - 2] = arr[i]
          
    arr[len(arr) - 1] = last
  
# Function to right rotate 
def right_rotate(arr):
      
    start = arr[len(arr) - 2] 
    for i in range(len(arr) - 4, -1, -2):
        arr[i + 2] = arr[i]
          
    arr[0] = start
  
# Function to rotate the array 
def rotate(arr):
      
    left_rotate(arr)
    right_rotate(arr) 
    for i in range(len(arr)): 
        print(arr[i], end = " ")
  
# Driver code 
arr = [ 1, 2, 3, 4, 5, 6 ] 
  
rotate(arr); 

File: Java/O(N)/ON_36.java
// Java program for above approach
import java.util.*;
class ON_36{
 
// Function to find no of iterations
static int minZero(int A[], int n)
{
 
    // Initialize count c = 0
    int c = 0;
 
    // If 0 already in array return c
 
    if (Arrays.stream(A).min().getAsInt()== 0)
        return c;
 
    // Iterate till we get zero in array
    while (Arrays.stream(A).min().getAsInt() != 0) {
 
        // Assign first element in x
        int x = A[0];
 
        // Loop to subtract consecutive element
        for (int i = 0; i < (n - 1); i++) {
            A[i] = Math.abs(A[i] - A[i + 1]);
        }
        A[n - 1] = Math.abs(A[n - 1] - x);
 
        // Increment count c
        c += 1;
    }
 
    // Return c
    return c;
}
 
// Driver Code
public static void main(String[] args)
{
   
    // Original array
    int arr[] = { 2, 6, 3, 4, 8, 7 };
 
    // Length of array
    int N = arr.length;
 
    // calling function
    int x = minZero(arr, N);
 
    // print the result
    System.out.print(x);
}
 
}

